### PR TITLE
AY profile generated at the end of installation contains valid hostname

### PR DESCRIPTION
**Pull request by mchf**
Bnc#960040

### CODE REVIEW

File: library/types/src/modules/Hostname.rb [hunk 1]
```diff
@@ -155,30 +155,20 @@ def MergeFQ(hostname, domain)
...
-
-        if fqhostname == "" || fqhostname.nil?
-          # last resort (#429792)
-          fqhostname = "linux.#{@DefaultDomain}"
-        end
-        Builtins.y2warning("Using fallback hostname %1", fqhostname)
+        Builtins.y2warning("Using fallback hostname")
+
+        fqhostname = SCR.Read(path(".target.string"), "/etc/hostname") || ""
+        fqhostname = "linux.#{@DefaultDomain}" if fqhostname.nil? || fqhostname.empty?
```
Comment (teclator) [inline library/types/src/modules/Hostname.rb]: NP: fqhostname.nil? is not needed

**Comment by teclator** [comment]
LGTM

**Comment by mvidner** [comment]
If you feel adventurous, you can even replace SCR with 
```rb
require "yast2/execute"
require "yast2/target_file"
Yast::Execute.on_target("hostname", "--fqdn")
Yast::TargetFile.read("/etc/hostname")
```

File: library/types/src/modules/Hostname.rb (hunk 1 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: ah sure, that was reason for the above change and in the end I forgot to delete it;-) Thanks for pointing that.

**Comment by mchf** [comment]
@mvidner: This function is used even in SCR chroot magic parts of code and similar, so I don't feel that brave yet ;-)

**Comment by mvidner** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for reviews.

**Comment by kobliha** [comment]
No testcase?

**Pull request merged**